Lib/typing.py

```diff
@@ -1083,6 +1083,10 @@ def _proto_hook(other):
 
             # Second, perform the actual structural compatibility check.
             for attr in _get_protocol_attrs(cls):
+                # Requiring people to implement this on e.g. iterable classes
+                # would break existing code
+                if attr == '__class_getitem__':
+                    continue
```

Comment:
Instead of adding this, you can add `__class_getitem__` to `EXCLUDED_ATTRIBUTES`, which exists for such exclusions.